word2vec.py

```diff
@@ -0,0 +1,73 @@
+import re
+import string
+
+import gensim
+import numpy as np
+import pandas as pd
+from tqdm import tqdm
+
+from pycocotools.coco import COCO
+
+
+def sentence2vec(sentence, model):
+    ''' Converts a sentence to a vector by averaging each word vector
+    representation given by the model '''
+    # Remove all non alphanumeric characters
+    regex = re.compile('[\W_ ]+') # TODO: compiling once for all calls might be faster
+    regex = re.compile('[%s]' % re.escape(string.punctuation))
```

Comment:
Je connais pas, en tout cas, selon ce post ça a l'air d'être une des approches la plus rapide:
http://stackoverflow.com/questions/265960/best-way-to-strip-punctuation-from-a-string-in-python